modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+  <http:listener doc:name="Listener"
+      outputMimeType="application/json; streaming=true"
+      config-ref="HTTP_Listener_config" path="/input"/>
+</flow>
+----
+
+Notice that `streaming=true` is part of the `outputMimeType` value. Other Mule components, such as the File and FTP components, also support the MIME Type setting.
+
+To pass the streamed output to the next message processor, you can use the `output` directive in a DataWeave script, for example:
+
+[source,XML,linenums]
+----
+output application/json deferred=true
+----
+
+For more detail on use of the `deferred` property, see <<stream_output>>.
+
+[[stream_csv]]
+== Streaming CSV
+
+CSV is simplest format for streaming because of its structure. Each row below the CSV header is a streamable record. The following CSV example consists of records that contain `name`, `lastName`, and `age` values.
+​
+[source,CSV,linenums]
+----
+name,lastName,age
+mariano,achaval,37
+leandro,shokida,30
+pedro,achaval,4
+christian,chibana,25
+sara,achaval,2
+matias,achaval,8
+----
+​
+To stream the CSV example, the following script selects values from each record. It uses the `map` function to iterate over each record in the document.
+​
+[source,DataWeave,linenums]
+----
+payload map (record) ->
+    {
+        FullName: record.lastName ++ "," ++ record.name,
+        Age: record.age
+    }
+----
+
+Although streaming does not support random access to the entire document, a DataWeave script can access data randomly _within each record_ because each record is loaded into memory. For example, the expression `record.lastName ++ "," ++ record.name,` can access a `lastName` value before it accesses a `name` value even though the order of values is reversed in the input.
+
+However, streaming does not work in the following script. The script requires random access to the entire document to return the elements in a different order than they are given in the input.
+​
+[source,DataWeave,linenums]
+----
+[payload[-2], payload[-1], payload[3]]
+----
+​
+[[stream_json]]
+== Streaming JSON
+​
+The unit of a JSON stream is each element in an array.
+
+Note that DataWeave added limited support for JSON streaming in version 2.2.0 for Mule 4.2. To stream JSON in that version, DataWeave required the root to be an array. In Mule 4.3, DataWeave extended support for streaming to arrays that are not at the root of the input.
+
+[[example_json]]
+[source,JSON,linenums]
+----
+{
+    "name" : "Mariano",
+    "lastName": "Achaval",
+    "family": [
+        {"name": "Sara", "age": 2},
+        {"name": "Pedro", "age": 4},
+        {"name": "Matias", "age": 8}
+    ],
+    "age": 37
+}
+----
+​
+DataWeave can stream `payload.family` and perform random access within each element of that array. However, DataWeave cannot randomly access the container object. For example, it is not possible to stream `{ a: payload.age , b: payload.family}` because `age` follows `family`, and DataWeave cannot go backwards.
```

Comment:
```suggestion
In this example, DataWeave can stream `payload.family` and perform random access within each element of that array. However, DataWeave cannot randomly access the container object. For example, it is not possible to stream `{ a: payload.age , b: payload.family}` because `age` follows `family`, and DataWeave cannot go backwards.
```